Tools/cases_generator/generate_cases.py

```diff
@@ -171,7 +172,6 @@ def effect_str(effects: list[StackEffect]) -> str:
                 popped, pushed = stacking.get_stack_effect_info_for_macro(instr)
             case parsing.Pseudo():
                 instr = self.pseudo_instrs[thing.name]
-                popped = pushed = None
```

Comment:
Mypy was eagerly narrowing the type of `pushed` here to be `None`, meaning that it was inferring the `else` branch on line 188 to be unreachable. Assigning to `popped` and `pushed` at the same point where their types are declared solves this problem.

Reply:
I think that you can revert the L186 code into the original one that I modified in the previous PR.

````diff
@@ -182,8 +182,7 @@ def effect_str(effects: list[StackEffect]) -> str:
                     assert target_instr
                     target_popped = effect_str(target_instr.input_effects)
                     target_pushed = effect_str(target_instr.output_effects)
-                    if pushed is None:
-                        assert popped is None
+                    if popped is None and pushed is None:
                         popped, pushed = target_popped, target_pushed
                     else:
                         assert popped == target_popped
````

Reply:
> Assigning to `popped` and `pushed` at the same point where their types are declared solves this problem.

Still, that feels unsatisfactory. Each case is supposed to set these variables, so the initialization at the declaration point ought to be unnecessary. It seems mypy is just wrong here?

Even though I agree that this is probably the best solution for now, in general I feel that there's a lot of churn in this piece of code due to limitations in mypy. :-(

Reply:
> Still, that feels unsatisfactory. Each case is supposed to set these variables, so the initialization at the declaration point ought to be unnecessary

Right, this PR reduces symmetry between the branches in a less-than-ideal way. I didn't see that at first (apologies, I'm new to this code).

I think there's an alternative solution, but I might not get to it for a day or two; I'm traveling this weekend.